assets/sharedClientFunctions.js

```diff
@@ -1,149 +0,0 @@
-        }
-      });
-    };
-    $scope.deleteRef = function (id) {
-      var url = "/reference/delete";
-      io.socket.post(url, {refId: id}, function (data, res) {
-        if (res.statusCode === 200) {
-          $scope[current_user].references = $scope[current_user].references.filter(function (ref) {
-            return ref.id !== id;
-          });
-          $scope.$apply();
-        } else {
-          console.log(res.statusCode + ": " + data);
-        }
-      });
-    };
-    $scope.isNotNormalTrade = referenceService.isNotNormalTrade;
-    $scope.hasNumber = referenceService.hasNumber;
-    $scope.isEvent = referenceService.isEvent;
-    $scope.isShiny = referenceService.isShiny;
-    $scope.isCasual = referenceService.isCasual;
-    $scope.isEggCheck = referenceService.isEggCheck;
-    $scope.isTrade = referenceService.isTrade;
-    $scope.isInvolvement = referenceService.isInvolvement;
-    $scope.isEgg = referenceService.isEgg;
-    $scope.isBank = referenceService.isBank;
-    $scope.isGiveaway = referenceService.isGiveaway;
-    $scope.isEggCheck = referenceService.isEggCheck;
-    $scope.isMisc = referenceService.isMisc;
-    $scope.isApproved = referenceService.isApproved;
-    $scope.getRedditUser = referenceService.getRedditUser;
-    $scope.formattedName = flairService.formattedName;
-    $scope.inPokemonTradesTrader = flairService.inPokemonTradesTrader;
-    $scope.inPokemonTradesHelper = flairService.inPokemonTradesHelper;
-    $scope.inSVExchangeHatcher = flairService.inSVExchangeHatcher;
-    $scope.inSVExchangeGiver = flairService.inSVExchangeGiver;
-    $scope.getFlair = flairService.getFlair;
-    $scope.userHasFlair = function (flair) {
-      return flairService.userHasFlair($scope.user, flair);
-    };
-    $scope.numberOfTrades = function () {
-      return referenceService.numberOfTrades($scope[current_user]);
-    };
-    $scope.numberOfPokemonGivenAway = function () {
-      return referenceService.numberOfPokemonGivenAway($scope[current_user]);
-    };
-    $scope.numberOfEggsGivenAway = function () {
-      return referenceService.numberOfEggsGivenAway($scope[current_user]);
-    };
-    $scope.numberOfEggChecks = function () {
-      return referenceService.numberOfEggChecks($scope[current_user]);
-    };
-    $scope.numberOfApprovedEggChecks = function () {
-      return referenceService.numberOfApprovedEggChecks($scope[current_user]);
-    };
-    $scope.getFlairTextForSVEx = function () {
-      return flairService.getFlairTextForSVEx($scope[current_user]);
-    };
-    $scope.applied = function (flair) {
-      return flairService.applied($scope.user, flair);
-    };
-    $scope.canUserApply = function (applicationFlair) {
-      return flairService.canUserApply($scope.user, applicationFlair || $scope.selectedFlair, $scope.flairs);
-    };
-    $scope.formattedRequirements = function (flair) {
-      return flairService.formattedRequirements(flair, $scope.flairs);
-    };
-  }
-};
```

Comment:
Alright. In that case, what should we do with the shared functions? There are a few options that I can think of:
1. Have a helper function in another file to add them to the scope, and call that function from *all* the controllers (this is what we were doing before, which was redundant since this helper function only needs to be called once).
2. Have a helper function in another file, but only call the helper function from one of the controllers.
3. Explicitly define all the functions in a controller that's always active (what I was doing here).
4. Add the functions to `$scope` in app.js using [run()](https://docs.angularjs.org/api/ng/type/angular.Module#run). This isn't a bad option, but it still causes functions to end up defined even if they're not needed. It also creates a strange situation where some of our functions are defined in controllers, and some are defined in app.js.

Personally I think 3 or 4 are best.